src/lib/net/src/connection.rs

```diff
@@ -159,15 +162,22 @@ pub async fn handle_connection(state: Arc<ServerState>, tcp_stream: TcpStream) -
         .instrument(debug_span!("eid", %entity))
         .inner()
         {
-            warn!(
-                "Failed to handle packet: {:?}. packet_id: {:02X}; conn_state: {}",
-                e,
-                packet_skele.id,
-                conn_state.as_str()
-            );
-            // Kick the player (when implemented).
-            terminate_connection(state.clone(), entity, "Failed to handle packet".to_string())
-                .await?;
+            match e {
+                NetError::Kick(msg) => {
+                    terminate_connection(state.clone(), entity, msg.clone())
```

Comment:
Perhaps adding an Ext trait to player for specifically terminating connection? So you can do: `entity.terminate_conn(state, msg)` instead of having to need to call a function might be a cleaner approach.

Reply:
> Perhaps adding an Ext trait to player for specifically terminating connection? So you can do: `entity.terminate_conn(state, msg)` instead of having to need to call a function might be a cleaner approach.

Maybe but I didn't add this function

Reply:
ext trait to the Entity (usize). `TerminateConnectionPlayerExt` for `usize`. or something, and add same signature with `(&self, state: GlobalState, message: impl Into<TextComponent>)` or whatever.

Reply:
> ext trait to the Entity (usize). `TerminateConnectionPlayerExt` for `usize`. or something, and add same signature with `(&self, state: GlobalState, message: impl Into<TextComponent>)` or whatever.

Well I also added it this way so that I can prevent other events running on a disconnected player